Py3 qt5

Valdas2000 opened the pull request:
More features
1 - Gentle cut feature. Allows to setup number of cuts and space
2 - Unit tests for metric
3 - New bit_gentle setup parameter (we got minor version grows for that)

Known issues:
Exact fit is red for some reason

TODO:
0 - Check formatting
1 - Ability to reverse variable space
2 - Change labels on stripe
3 - Show perfect fit depth for dovetails

Valdas2000 committed:
[~] imperial for metric. Initial changes

Valdas2000 committed:
[~] Spacing improvment

angle=0 bits tested. Spacing and width fixed.
Need to test dovetails

Valdas2000 committed:
[~] fix centred

Centred cuts seeems fixed

Valdas2000 committed:
[~] Fixed non centred cuts

Non centred cuts fixed.
Known issues are
1- variable cuts
2- no markers ineditor

Valdas2000 committed:
[~] fix variable spasing for integer diameters.

Valdas2000 committed:
[~] Variablespasing fix

To do:
1 - fix extra calculations
2 -fix redraw
3 - fix 3 and 4 board cuts

Valdas2000 committed:
[~] Commit more changes

Valdas2000 committed:
[~] midline inequal spacecuts

also 2and 3 boards fix

Valdas2000 committed:
[~] Fix for Equal and Variable cuts. 

Width slider fixed to show actual cut width
Cuts works fine now for any number of boards
Fixed precision issues
Fixed Finger Width display (f.3 used from now)
TODO: 
1. Fix add cut in editor
2. Fix move and change size precision issues

Valdas2000 committed:
[~] Minor changes. Symmwtry improved

Valdas2000 committed:
[~] Variable spacing move fixed

Valdas2000 committed:
[~] Variable spacing fixed

Valdas2000 committed:
[~] Add Fixed

Seems all corner cases covered now
TODO: Verify imperial revision

Valdas2000 committed:
[~] Variable spacing add changes

Valdas2000 committed:
[~] Minor fixed to get imperial measurments back to work

1 - Fix bit width processing
2 - fix QT5 issues in unit tests
TODO
Fix issues discovered in unit tests

Valdas2000 committed:
[~] Fix Screenshot (QT5 Migration)

Valdas2000 committed:
[~] Variable spasing fix

TODO: fix compute_fit issue

Valdas2000 committed:
[+] Fix Mouse issues (QT5)

Valdas2000 committed:
[+] Fix Gap Calc

TODO: Final review/ Refactor the code

Valdas2000 committed:
[+] Cutpass

Valdas2000 committed:
[~] Gentle cut completed

1-New optionin setup
2- Minorversion updated

Valdas2000 committed:
[+] Unit tests for metric

Valdas2000 committed:
Merge branch 'Imperic-DT-in-metric' into py3_qt5

lowrie commented on router.py:
```diff
@@ -109,6 +110,23 @@ def __init__(self, units, width, depth, angle=0):
+
+    def set_gentle_from_string(self, s):
+        '''
+        Sets the width from the string s, following requirements from units.string_to_increments().
```
fix comment

Valdas2000 replied:
Hello. The point is that gentle value in present (%). So no increment conversion required

lowrie commented on config_file.py:
```diff
@@ -268,6 +272,172 @@
                'separation':6}
 
 
+# values that are migrated to new versions of the config file.  Don't include metric, because it's
```
It's not clear to me why so much code changed here.  Any idea?

Valdas2000 replied:
Ghmm.
The file content dobbled (repeated 2 times).

lowrie replied:
Ahh, sorry I missed that.

lowrie commented on config_file.py:
```diff
@@ -118,6 +118,9 @@
 # If the overlap in the joint exceeds this value, warn the user [inches|mm]
 warn_overlap = {warn_overlap}
 
+# Cutting part of the bit %
```
I don't understand "cutting part of the bit".  Why is that different than width?

Valdas2000 replied:
Let we have straight 1/2" bit. If we do the full width cut on the edge of out board we can chip the wood. So it's better to make more passes to cut a smaller parts (usually a half of bit width which is 1/4 in our case or 50%)
Same is about internal cuts. We can make passes to get more clean cut. So the parameter is to tell how much wood we cut off in a single pass.

Valdas2000 replied:
Let we need to make 1" grove with a 1/2" bit. We can make 2 passes but the center of grove would not be clean (that means gentle = 100%).  In case we made 3 cuts we got fine surface (we can made even more but it does not make much sense). So we can tell program to avoid full size cuts if possible. 1st cut is full size in any case. For the next cuts application will take "gentle into accout) and shift place next cut to value less then 1/2 (in case of gentle = 50% the shift will be 1/4") the 3rd (2st + 1/4") cut will finalize grove.

Valdas2000 replied:
Width means the width of grove not the width of a bit.
My English is not that good to delivery this tiny difference in comments.

lowrie replied:
I understand, thank you.  And your English is a lot better than my Russian.

lowrie commented:
@Valdas2000 I am very busy with my real job, but I will try to get this merged over the weekend.

Valdas2000 committed:
[~] Depricated imp updated to importlib (Python 3 compatibility update)

lowrie commented:
@Valdas2000 The code locks up if my config file (`~/.pyrouterjig`)  sets `metric = False`, and I run `python qt_test_metric.py`.  Can you fix this?

Valdas2000 commented:
oops
I put metric config restriction into the test. 


On Sun, Feb 18, 2018 at 5:39 PM, Robert B. Lowrie <notifications@github.com>
wrote:

> @Valdas2000 <https://github.com/valdas2000> The code locks up if my
> config file (~/.pyrouterjig) sets metric = False, and I run python
> qt_test_metric.py. Can you fix this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lowrie/pyRouterJig/pull/5#issuecomment-366520683>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AhlUv1-uoGZOlNyF0XVM5GxRpe5p_Dzsks5tWDYugaJpZM4SCzTp>
> .
>



-- 
Regards
Valentin

Valdas2000 committed:
Fail all tests in case of incorrect units specifyed in config.

Halt happen because the application is unable to calculate proper variable cuts for wide-wide-wide boards.
Calculation issue will be fixed in upcoming pull request

Valdas2000 committed:
[+] initial version of translation file

Localization is not committed yet but the file works fine on my local Windows pc.
The question is if it is possible to compile it on MAC

Valdas2000 committed:
[~] Updated translation. Menu, messages, settings, combos

Valdas2000 committed:
[+] Multilanguage support

[+] - language param in config file
[+] - application load qm file from ts subfolder
[+] - interfaice, messages, exceptions translated
[+] - combo box lists are translated. The way of combos handled changed. Now we are working with itemData because text is vary
[~] - default_wood now saves the pattern ID or file instead of the name
[+] - color selection fixed in property menu
[~] - disable metric and English scale mix. Load Metric file into English and reverse is not allowed

Valdas2000 committed:
Merge pull request #4 from Valdas2000/Valdas2000-multilang

[+] Multilanguage support

Valdas2000 committed:
Translate for slider lables

Valdas2000 committed:
[+] Slider translation, board width limitation

[+] Board width limited to 1 meter or 4 feet
[+] Slider translation 
[~] Minor fixes in Status line translation

Valdas2000 committed:
[!] Critical fix. Precission and rounding fixed for 4 feet board limits

Valdas2000 committed:
Merge pull request #5 from Valdas2000/Valdas2000-patch-2

[!] Critical fix. Precission and rounding fixed for 4 feet board limits

Valdas2000 committed:
[~] Typo

Valdas2000 committed:
[+] Board_with range checker

Prevent switch to variable spacing for narrow board
Prevent set narrow board width for variable spacing

Valdas2000 committed:
[~] Board_with range checker

Prevent set narrow board width for equal spacing

Valdas2000 committed:
[~] Updated translation for updated titles

Valdas2000 committed:
[+] updated titles

Title got updated format

Valdas2000 committed:
Merge pull request #6 from Valdas2000/Valdas2000-patch-1

[+] updated titles

Valdas2000 committed:
[~] Status line fixed

Valdas2000 committed:
[+] Add Language switch into setup dialog

Valdas2000 committed:
Merge pull request #7 from Valdas2000/Valdas2000-patch-1

[+] Add Language switch into setup dialog

Valdas2000 committed:
[~] Metric Test Fixed

With updated precission  the value rounded to 1/100. Prior releases was not implement perfect rounding.

Valdas2000 committed:
[~] Python3.  Depricated imp removed. importlib used

Valdas2000 committed:
[+] Variable options

lowrie commented:
I am getting 1 error in both qt_test and qt_test_metric.  For qt_test_metric:

======================================================================
ERROR: test_variable_spaced (__main__.Driver_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "qt_test_metric.py", line 304, in test_variable_spaced
    self.d._on_cb_vsfingers(i)
  File "/Users/lowrie/pyRouterJig/repo/qt_driver.py", line 1316, in _on_cb_vsfingers
    self.draw()
  File "/Users/lowrie/pyRouterJig/repo/qt_driver.py", line 998, in draw
    self.description)
  File "/Users/lowrie/pyRouterJig/repo/qt_fig.py", line 223, in draw
    self.config)
  File "/Users/lowrie/pyRouterJig/repo/router.py", line 759, in __init__
    cut_boards(boards, bit, spacing)
  File "/Users/lowrie/pyRouterJig/repo/router.py", line 726, in cut_boards
    boards[0].set_bottom_cuts(last, bit)
  File "/Users/lowrie/pyRouterJig/repo/router.py", line 363, in set_bottom_cuts
    c.make_router_passes(bit, self)
  File "/Users/lowrie/pyRouterJig/repo/router.py", line 607, in make_router_passes
    self.validate(bit, board)
  File "/Users/lowrie/pyRouterJig/repo/router.py", line 584, in validate
    'Must have xmax > xmin!') % (self.xmin, self.xmax))
router.Router_Exception: cut xmin = 0, xmax = -4: Must have xmax > xmin!

lowrie commented:
The `cases` variable in `qt_test_metric.py` use imperial arguments for `width`.  Shouldn't these be metric widths?

lowrie commented:
In `qt_test.py`, the `Driver_Test.setUp()` function needs:

```
# the application must be not be in metric mode for these tests
self.assertTrue(self.d.config.metric == False)
```

Valdas2000 commented:
You right.
The English scale tests should not allow run in Metric scales.
The point is that I did not commit it.
I updated qt_test.py updated my sources but they are ahead of commit.

Valdas2000 committed:
[~] Disable test in case of Metric cinfiguration

Valdas2000 committed:
[~] Fix Variable spasing issue for normal values

Valdas2000 commented:
Seems I have no features to add into existent functionality.

We need more testing for sure. Existent tests have to be extended to cover modern functionality.
Issues:
1 - 3DS. I have no way to check the file
2 - Need to fix compatibility for Variable space and Edited space files saved in previous. (can be fixed)

Ideas:
1- Join jig calculation. (Feature to make templates for rings)

lowrie commented:
@Valdas2000 Sorry for the slow response.  `qt_test.py` now has 6 errors when I do the following:
 
```
rm ~/.pyrouterjig # or move it
python qt_test.py # then select English units.
```
Similarly, for metric units, to run the tests I do:
```
rm ~/.pyrouterjig # or move it
python qt_test_metric.py #  then select metric units.
```
Metric has 0 errors, which is good.  Please test both unit systems as above.

Don't worry about 3DS.  Rings sounds interesting!  I've never made any woodworking project with rings.

Valdas2000 commented:
Well. There are 3 cases:

1 - In case there is no config found the application show the dialog to select units.
However our tests have no ability to setup proper units. 
I think it is OK if they failed.

2 - In case the old config found the tests may fail. 
I have no idea what is the proper test behavior in that case

3 - Modern config exists
There are no issues with it.

BTW
The last commits  contain updates for the config. I did not grow the version up so so it better to delete config if there was any runs on versions earlier than combo boxes update.

lowrie commented:
@Valdas2000 No, your summary of the code's behavior for these cases is incorrect.  Here's what should happen:

Case 1 (no config found): The user selects the units in the dialog, and then a default config file is then created.  If you're running `python qt_test.py` and select `English` in the dialog, I get 6 serious errors.  This is the problem.  This case should not cause errors.

Case 2a (old config with English units, `metric = False`): Then a modern config is created from the old config, with English units. Again, `python qt_test.py` will likely create 6 errors, but I have not tested this case.

Case 2b (old config with metric units, `metric = True`):  Then a modern config is created from the old config, with metric units. `python qt_test.py` will likely work, in the sense that it does not run any tests because the units are wrong.

Case 3a (modern config with`metric = False`): Then `python qt_test.py` creates 6 errors; I have tested this case to confirm this behavior.

Case 3b (modern config with`metric = True`): Then `python qt_test.py` works, for the same reason as Case 2b.

Valdas2000 committed:
[~] Fixed direct access to config value with no conversion

Valdas2000 commented:
Seems the direct access to non converted config value was the cause of issue.
There is one more thing. 
Variable spaced files from previous version got different spacing

lowrie commented:
@Valdas2000 Excellent work!
